lib/construct.ts

```diff
@@ -76,10 +75,6 @@ export class Node {
       // This is a root construct.
       this.id = id;
     }
-
-    if (Token.isUnresolved(id)) {
```

Comment:
I wonder if we should allow users to pass in an `identityValidator` (similar to `nodeFactory`) so we can implement this in the AWS CDK to protect against using tokens in construct IDs?

Reply:
Sounds interesting... let me look into it.